server/charge.js

```diff
@@ -18,45 +18,140 @@ exports.charge = function() {
 
       if (Date.now() > left) {
         admin.database().ref('/users/' + snap[key].userUid).once('value').then((snapshot) => {
-          
-          const authStr = 'Basic ' + btoa(snapshot.val().settings.togglKey + ':api_token')
           const time = moment(snap[key].startedAt).format('YYYY-MM-DD')
-          axios({
-            headers: {
-              Authorization: authStr
-            },
-            url: 'https://toggl.com/reports/api/v2/details',
-            params: {
-              user_agent: snapshot.val().settings.email,
-              workspace_id: snapshot.val().settings.workspace,
-              user_ids: snapshot.val().settings.user_ids,
-              since: time
-            }
-          })
-          .then(response => {
-            if (response.data.total_grand - snap[key].timeWorkedBefore < snap[key].hours * 60 * 60 * 1000) {
 
-              stripe.charges.create({
-                amount: snap[key].money * 100,
-                currency: "usd",
-                description: "Example charge",
-                customer: snapshot.val().customerId
-              }, function(err, charge) {
-                if (err) {
-                  console.log(err)
-                } else {
-                  console.log('Chargeeeeee')
-                }
-              })
-              .then(() => {
-                admin.database().ref('/days/' + key).update({
-                  isCharged: true,
-                  isDayOver: true
+          if (snapshot.val().togglKey && snapshot.val().email && snapshot.val().workspace && snapshot.val().user_ids && snapshot.val().hubstaffAuthToken) {
```

Comment:
next 130 lines can be refactored to 50 lines
don't these if cases look familiar?